plugin.cmake

```diff
@@ -1,75 +1,35 @@
-add_minerva_python_style_test(rest "server/rest")
-add_minerva_python_style_test(utility "server/utility")
-add_minerva_python_style_test(bsve "server/utility/bsve")
-add_minerva_python_style_test(jobs "server/jobs")
-
-add_minerva_eslint_test(minerva "web_external/js")
-add_minerva_eslint_test(minerva-gruntfile "Gruntfile.js")
+add_standard_plugin_tests(NO_SERVER_TESTS NO_CLIENT_TESTS)
+add_python_test(analysis PLUGIN minerva)
+add_python_test(dataset PLUGIN minerva)
+add_python_test(geojson PLUGIN minerva)
+add_python_test(import_analyses PLUGIN minerva)
+add_python_test(session PLUGIN minerva)
+add_python_test(twofishes PLUGIN minerva)
+add_python_test(wms PLUGIN minerva)
+
+set_property(TEST python_static_analysis_minerva PROPERTY LABELS minerva_server)
+set_property(TEST server_minerva.analysis PROPERTY LABELS minerva_server)
+set_property(TEST server_minerva.dataset PROPERTY LABELS minerva_server)
+set_property(TEST server_minerva.geojson PROPERTY LABELS minerva_server)
+set_property(TEST server_minerva.import_analyses PROPERTY LABELS minerva_server)
+set_property(TEST server_minerva.session PROPERTY LABELS minerva_server)
+set_property(TEST server_minerva.twofishes PROPERTY LABELS minerva_server)
+set_property(TEST server_minerva.wms PROPERTY LABELS minerva_server)
 
 add_web_client_test(
     minerva "${PROJECT_SOURCE_DIR}/plugins/minerva/plugin_tests/client/minervaSpec.js"
-    ENABLEDPLUGINS "gravatar" "jobs" "minerva"
-    BASEURL "/static/built/testEnvMinerva.html"
-    TEST_MODULE "plugin_tests.web_client_test"
+    PLUGIN minerva
+    ENABLEDPLUGINS "gravatar" "jobs"
+    SETUP_MODULES "${_pluginDir}/plugin_tests/create_user.py"
 )
-set_property(TEST web_client_minerva PROPERTY LABELS minerva_client)
+set_property(TEST web_client_minerva.minerva PROPERTY LABELS minerva_client)
 
 add_web_client_test(
-    geojson "${PROJECT_SOURCE_DIR}/plugins/minerva/plugin_tests/client/geojsonUtil.js"
-    BASEURL "/static/built/testEnvMinerva.html"
+    geojson "${PROJECT_SOURCE_DIR}/plugins/minerva/plugin_tests/client/geojsonUtilSpec.js"
+    PLUGIN minerva
+    ENABLEDPLUGINS "gravatar" "jobs"
 )
-set_property(TEST web_client_geojson PROPERTY LABELS minerva_client)
-
-set_property(TEST js_coverage_reset APPEND PROPERTY LABELS minerva_client)
-set_property(TEST js_coverage_combine_report APPEND PROPERTY LABELS minerva_client)
+set_property(TEST web_client_minerva.geojson PROPERTY LABELS minerva_client)
 
-set_property(TEST py_coverage_reset APPEND PROPERTY LABELS minerva_server)
-set_property(TEST py_coverage_combine APPEND PROPERTY LABELS minerva_server)
-set_property(TEST py_coverage APPEND PROPERTY LABELS minerva_server)
-set_property(TEST py_coverage_html APPEND PROPERTY LABELS minerva_server)
-set_property(TEST py_coverage_xml APPEND PROPERTY LABELS minerva_server)
+set_property(TEST puglint_minerva PROPERTY LABELS minerva_client)
+set_property(TEST eslint_minerva PROPERTY LABELS minerva_client)
```

Comment:
Let's add puglint of the web_external, too:
```
add_puglint_test(minerva_external "${CMAKE_CURRENT_LIST_DIR}/web_external/templates")
set_property(TEST puglint_minerva_external PROPERTY LABELS minerva_client)
```
This will show some additional puglint errors which should be fixed.